src/main/java/com/_up/megastore/services/implementations/CategoryService.java

```diff
@@ -37,4 +37,28 @@ private Category getSuperCategory(UUID superCategoryId) {
         return superCategoryId != null ? findCategoryByIdOrThrowException(superCategoryId) : null;
     }
 
+    @Override
+    public CategoryResponse deleteCategory(UUID categoryId){
+        ifCategoryExistThrowException(categoryId);
+        ifCategorySubcategoriesExistThrowException(categoryId);
+        Category category = ifCategoryIsNotDeletedThrowException(categoryId);
+        category.setDeleted(true);
+        return CategoryMapper.toCategoryResponse(categoryRepository.save(category));
+    }
```

Comment:
Esto podría optimizarse para pegarle sólo una vez a la bd

```suggestion
    public CategoryResponse deleteCategory(UUID categoryId){
        Category category = findCategoryByIdOrThrowException(categoryId);
        ifCategorySubcategoriesExistsThrowException(category);
        category.setDeleted(true);
        categoryRepository.save(category);
    }
```